platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/FelixConfig.java

```diff
@@ -0,0 +1,111 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+import org.osgi.service.cm.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for enabling functional constructs with {@link Configuration}s and exposes the
+ * felix.fileinstall.filename property, if it exists, as a {@link File}.
+ */
+public class FelixConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FelixConfig.class);
+
+    private static final String FELIX_FILENAME_PROP = "felix.fileinstall.filename";
+
+    private Configuration config;
+
+    private File felixFile;
+
+    FelixConfig(Configuration config) {
+        this.config = config;
+        this.felixFile = extractFelixFileInstallProp(config);
+    }
+
+    FelixConfig(Supplier<Configuration> configSupplier) {
+        this(configSupplier.get());
+    }
+
+    /**
+     * @return the pid for this configuration.
+     */
+    String getPid() {
+        return config.getPid();
+    }
+
+    /**
+     * @return the File object pointing to the this configuration on disk.
+     */
+    @Nullable
+    File getFile() {
+        return felixFile;
+    }
+
+    /**
+     * Determine if the felix.fileinstall.filename property changed.
+     *
+     * @param original file location to compare to the current file location poitned to by this
+     *                 config.
+     * @return False if the file locations by absolute path are identical. True otherwise.
+     */
+    boolean filePropChanged(File original) {
+        return !Objects.equals(felixFile, original);
+    }
+
+    private File extractFelixFileInstallProp(Configuration config) {
+        if (config.getProperties() == null) {
+            LOGGER.debug("Config props dictionary is null, was the config just created?");
+            return null;
+        }
+        Object felixConfigFileName = config.getProperties()
+                .get(FELIX_FILENAME_PROP);
+        if (felixConfigFileName == null) {
+            LOGGER.debug("Felix filename prop not found for pid {}", config.getPid());
+            return null;
+        }
+        return createFileFromFelixProp(felixConfigFileName);
+    }
+
+    private File createFileFromFelixProp(Object felixConfigFileName) {
```

Comment:
I prefer a more functional approach since the technique you mentioned makes stateful assumptions during object initialization.